packages/block-library/src/navigation/edit.js

```diff
@@ -145,19 +148,24 @@ function Navigation( {
 
 	const placeholder = useMemo( () => <PlaceholderPreview />, [] );
 
+	// When the block is selected itself or has a top level item selected that
+	// doesn't itself have children, show the standard appender. Else show no
+	// appender.
+	const appender =
+		isSelected ||
+		( isImmediateParentOfSelectedBlock && ! selectedBlockHasDescendants )
+			? undefined
```

Comment:
It turns out that `InnerBlock.DefaultAppender` isn't a thing, it resolves as `undefined`, so I've just put `undefined` here 😆